### DISCUSSION TITLE
Move anchorBody to "left" of heading text

### DISCUSSION THREAD

**Discussion by ritchiecarroll** [allejo/jekyll-anchor-headings#20] [Q&A] [answered]
This is a proposed feature request:

Using the following Gist creates GitHub markdown style headers with anchors: https://gist.github.com/ritchiecarroll/132715b38c071cbd01b1d3a0da60822a

Like so:
![image](https://user-images.githubusercontent.com/9202752/88100255-ea379980-cb61-11ea-86de-d500408f74de.png)

If the anchor body was positioned to the left of the heading text, the visible icon would the appear on the left - similar to how it looks in GitHub markdown.

Could just make it an option? Like `anchorBodyPosition="left"`.

Not too familiar with pure liquid, but would be neat maybe to have an option like `useOctiIconAnchors="true"` (or something) to do the the whole shebang, i.e., left align, use svg octicon, add style to auto-show on hover.

Ritchie

**Comment by allejo** [comment, marked as answer]
There is a `beforeHeading` parameter you can pass to `anchor_headings.html` and it'll place the anchor before the heading. Then you can position it with CSS so it doesn't push the heading over to the right.

As for introducing octicon support, definitely something worth considering! Not sure if that'd be a bit too opinionated but it would make it easier for users without much experience customizing things.

**Comment by ritchiecarroll** [comment]
> There is a `beforeHeading` parameter you can pass to `anchor_headings.html` and it'll place the anchor before the heading. Then you can position it with CSS so it doesn't push the heading over to the right.

Thanks! I will give it a shot.

**Comment by ritchiecarroll** [comment]
Success! Thanks for the hint: https://gridprotectionalliance.github.io/go2cs/

Gist updated: https://gist.github.com/ritchiecarroll/132715b38c071cbd01b1d3a0da60822a

**Comment by allejo** [comment]
I've created a Wiki and added a page for examples from users. I've added a link to your Gist and copied your code snippets, @ritchiecarroll! I hope you don't mind :smile:

https://github.com/allejo/jekyll-anchor-headings/wiki/Examples#github-style-octicon-links
